rfcs/20250618-pr-rules/README.md

```diff
@@ -0,0 +1,207 @@
+# oneDAL Pull Request Rules
+
+## Introduction
+
+Generating, reviewing and merging code in oneDAL are key pillars in providing
+a high quality and performant product.  The oneDAL codebase is both large and
+intricate requiring domain knowledge and precision.  In order to guarantee these
+qualities in oneDAL for all contributors, a set of rules should be added.
+
+
+## Proposal
+
+This enumerates in detail the aspects and proceedures necessary to getting a
+Pull Request (PR) merged into oneDAL.  This (as listed in the preamble below)
+will assist new developers as well as standardize for all developers in
+completing the necessary steps. The goal would be to add this to oneDAL's
+public documentation and refer to it in the template (as a link) to all future
+Pull Requests to oneDAL.
+
+# Text to be added:
+
+## Preamble:
+
+This is the rules for merging PRs into the oneDAL code base on Github. This
+gives the minimum requirements for PR submitters and reviewers. The goal is
+to enumerate the steps for better efficiency in oneDAL development.
+
+## Section 1: Opening a PR, Ready for Review
+
+1. PR Template and Task Checklist
+   * All PRs must use the provided template, including the task checklist.
+2. Draft PRs
+   * PRs should initially be opened as Draft.
+3. Marking PRs as Ready for Review
+   * Before marking a PR as Ready for Review:
+     * All tasks in the checklist must be completed.
+     * Features and changes must be fully implemented.
+     * Both private and public CI pipelines must pass, or clear justification for any
+       failing tests must be provided.
+4. Reverting a PR Back to Draft
+   * Once a PR is converted from Draft to Ready for Review, it should only be
+     reverted to Draft if:
+     * Significant changes or new feature requests arise during the review.
+     * The requested changes depend on future dependencies (e.g., waiting for a new
+       release) or require substantial time to implement.
+     * Other blockers prevent the changes from being merged into the current main branch.
+
+## Section 2: General Rules
+
+1. A PR cannot introduce new failures in GitHub checks (public CI) (i.e. stays
+green) or new failures in private CI (pre-commit\*).
+2. Removal of CI tests can only be allowed with the written and documented
+consent\*\* of the code owners in extreme circumstances.
+3. Code which causes new CI failures or problems should apply additional
+fixes within a day, otherwise the PR should be reverted. 
+4. Code scans often do not run with PRs, as such merged PRs should not increase
+the number of issues for the Intel internal code scans\*\*\*, and may be
+grounds for a reversion.
+5. A merged PR which causes a code scan failure should be fixed before the next
```

Comment:
I wouldn't be focusing on Intel internal scans  and would write this as generic  code scans. Some of them are public, while others not yet. 

I think from internal scans at this point we have Coverity for oneDAL - which will be migrated to public instance.
Second one is protex IP - this one is generally usefull as it would catch potential not attributed contributions and license conflicts. I'm not sure if we have open alternatives for this.